File: mikan.scala
import scala.io.StdIn._

object Main extends App {
    val Array(n, m) = readLine().split(" ").map(_.toInt)

    def mikan(w: Int): Int = {
        Math.max((w + n / 2) / n, 1) * n
    }

    for (_ <- 0 until m) {
        println(mikan(readInt()))
    }
}

File: mikan.swift
import Foundation

let nm = readLine()!.split(separator: " ").map { Int($0)! }
let n = nm[0]
let m = nm[1]

func mikan(w: Int) -> Int {
    return max((w + n / 2) / n, 1) * n
}

for _ in 0..<m {
    print(mikan(w: Int(readLine()!)!))
}